### PR TITLE
feat: use LanceDB to manage generated models

### PR DESCRIPTION
**Pull request by kimusaku**
Here is the usage:

```python
import pandas as pd
from sapientml import SapientML, GeneratedModel
from sklearn.model_selection import train_test_split

train_data = pd.read_csv("housing-prices.csv")
train_data, test_data = train_test_split(train_data, train_size=0.9)
y_test = test_data["SalePrice"].reset_index(drop=True)
test_data.drop(["SalePrice"], axis=1, inplace=True)

sml = SapientML(["SalePrice"])

sml.fit(train_data, ignore_columns=["Id"])
model = sml.model

# export generated code 
model.export("outputs")

# save the model to LanceDB
id = model.save()

# load a model from LanceDB
model = GeneratedModel.load(id)


sml = SapientML.from_pretrained(model)
y_pred = sml.predict(test_data)

print(pd.concat([y_pred["SalePrice"].rename("SalePrice_pred"), y_test], axis=1))
```

### CODE REVIEW

**Review by AkiraUra — COMMENTED** [review]
- It seems weird to me that `save()` methods are different between lancedb and localfile. I prefer adding another method to store the data to lancedb such as `store()`.
- I think we can unify the two `GeneratedModel` classes because their contents are almost similar and the difference such as `id` and `timestamp` can be also included in `localfile.GeneratedModel`.  We can regard `id` as a state of the model.
- I think `id` should be determined in the last of `fit()` because the state is defined by `fit()`. That is, the state of the model is not changed until `fit()` is called again.
- I think we should provide a way for users to specify the URI of the database of LanceDB.
- I don't think GeneratedModel contain `score` and `metric` because the class is for the portability of the model, that is, so that users can call `predict()`. If we want to store the history of candidates' executions, we should provide another way for it.
